File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.js
/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(numbers, target) {
    let hashtable = new Map()
    for (let i = 0; i < numbers.length; i++){
        const r = target - numbers[i]
        if (hashtable.has(r) && hashtable.get(r) !== i) return [hashtable.get(r)+1,i+1]
        hashtable.set(numbers[i], i)
    }
    return [0, 0]
};

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
    
        int left = 0;
        int right = numbers.length  - 1;

        while (left < right){
            int s1 = numbers[left];
            int s2 = numbers[right];
            int sum = s1 + s2;
            if (sum > target){
                right = right - 1;
            } else if (sum < target) {
                left = left + 1;
            } else if (sum == target) {
                return new int[] {left+1, right+1};             
            }
        }
        return new int[] {-1, -1};
    }
}